meta-ostro/lib/image-dsk.py

```diff
@@ -1,56 +1,105 @@
+         " add it to the parameters." %
          (varname, ("used in " + location) if location else ""))
 
+
 def symlink(src, dst):
-    ''' Helper for symlink housekeeping. '''
+    """Helper for symlink housekeeping."""
     if os.path.exists(dst):
         os.remove(dst)
     os.symlink(src, dst)
 
-# Uses a raw partition generated elsewhere.
+
 def populate_rawcopy(src, dst):
+    """Argument src is a raw partition generated elsewhere."""
     shutil.copyfile(src, dst)
 
-# Creates and populates a FAT partition, out of a root directory.
+
 def populate_vfat(src, dst):
+    """Create and populate a FAT partition, out of a root directory <src>."""
     check_call(['mkdosfs', dst])
     check_call(['mcopy', '-i', dst, '-s'] + glob(src + '/*') + ['::/'])
 
-# Creates and populates an ext4 partition out of a root directory.
+
 def populate_ext4(src, dst):
+    """Create and populate an ext4 partition out of a root directory <src>."""
     check_call(['mkfs.ext4', '-F', dst] + (['-d', src] if src else []))
 
-def expand_vars(string, location=None):
-    return sub(r'\$\{([^}]+)\}', lambda x: lookup_var(x.group(1), location), string)
 
-# Entry point for generating the disk image.
+def expand_vars(arg_string, location=None):
+    """Expand variables in arg_string."""
+    return sub(r'\$\{([^}]+)\}', lambda x: lookup_var(x.group(1), location),
+               arg_string)
+
+
+def truncate_mib(fname, fsize):
+    """Create sparse file with requested size (in MiB)."""
+    with open(fname, "a+b") as fobj:
+        fobj.truncate(int(fsize) * 1024 * 1024)
+
+
+def sparse_copy(src_fname, dst_fname, offset_mib=0):
+    """Efficiently copy sparse file to or into another file."""
+    filemap = Filemap.filemap(src_fname)
+    try:
+        dst_file = open(dst_fname, 'r+b')
+    except IOError:
+        dst_file = open(dst_fname, 'wb')
+
+    for first, last in filemap.get_mapped_ranges(0, filemap.blocks_cnt):
+        start = first * filemap.block_size
+        end = (last + 1) * filemap.block_size
+
+        filemap._f_image.seek(start, os.SEEK_SET)
+        dst_file.seek((offset_mib * 1024 * 1024) + start, os.SEEK_SET)
+
+        chunk_size = 1024 * 1024
+        to_read = end - start
+        read = 0
+
+        while read < to_read:
+            if read + chunk_size > to_read:
+                chunk_size = to_read - read
+            chunk = filemap._f_image.read(chunk_size)
+            dst_file.write(chunk)
+            read += chunk_size
+    dst_file.close()
+
+
 def do_dsk_image():
+    """Entry point for generating the disk image."""
     # Load the descripton of the disk layout.
     partition_table = json.loads(expand_vars('${DSK_IMAGE_LAYOUT}'))
 
     # Before adding up the size of each partition, add the size of the GPT
     full_image_size_mb = partition_table["gpt_initial_offset_mb"] + \
-                         partition_table["gpt_tail_padding_mb"]
+        partition_table["gpt_tail_padding_mb"]
```

Comment:
nitpicking, iirc this sort of indentation is superseded by the one being replaced